src/services/order/order.service.ts

```diff
@@ -0,0 +1,116 @@
+import { CreateOrderDto } from '@dtos/order';
+import { User, UserAddress } from '@entities/auth';
+import { Cart } from '@entities/cart';
+import { Order, OrderItem } from '@entities/order';
+import { PaymentStatus } from '@enums/order';
+import { ConflictException, Logger, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { UserRepository } from '@repository/auth';
+import { CartRepository } from '@repository/cart';
+import { OrderRepository } from '@repository/order';
+import { ProductService } from '@services/product';
+
+export class OrderService {
+  private logger: Logger = new Logger('OrderService', {
+    timestamp: true
+  });
+
+  constructor(
+    @InjectRepository(OrderRepository)
+    private orderRepository: OrderRepository,
+    @InjectRepository(CartRepository)
+    private cartRepository: CartRepository,
+    @InjectRepository(UserRepository)
+    private userRepository: UserRepository,
+    private productService: ProductService
+  ) {}
+
+  async createOrder(
```

Comment:
For this particular function I see that you have multiple `await` calls, that will block the even-loop until the sequence of Promises are resolved, if you can a better approach would be use Promise.all or Promise.allSettled and start all those operations in parallel.